CONTRIBUTING.md

```diff
@@ -0,0 +1,149 @@
+# Contributing to crun
+
+Thanks for your interest in contributing!
+
+## Note: Before writing a big patch
+
+If you plan to contribute a large change, please get in touch *before*
+submitting a pull request by e.g. filing an issue describing your proposed
+change. This will help ensure alignment.
+
+## Background knowledge
+
+You will need to understand C to contribute to the core. There is also
+some Rust (more on this below).
+
+## Development environment
+
+Crun at its core is a low-dependency C library and CLI tools. You'll
+need a Linux environment, which could be a container or a VM/physical system.
+
+crun should be buildable on nearly any relatively modern Linux OS/distribution.
+
+### Building and testing
+
+crun uses [GNU Autotools](https://www.gnu.org/software/automake/manual/html_node/Autotools-Introduction.html) for its build system. Ensure that Autotools is installed and properly configured on your system.
+
+#### Setup
+
+To set up the build, run the following commands in the root directory:
+
+```bash
+./autogen.sh
+./configure
```

Comment:
The trap with this is we almost always want `--prefix=/usr` and the platform-specific proper libdir if it's going to be installed, but...maybe we can ignore that.